docs/troubleshooting/agent.md

```diff
@@ -0,0 +1,79 @@
+# The Switch Agent
+
+## Agent Overview
+
+The agent is a process authored by Hedgehog that creates and enforce the
+switch configurations. To view the status of every switch agent:
+
+```console
+kubectl get -o wide agent
+core@control-1 ~ $ kubectl get -o wide agents
+NAME           ROLE          DESCR         HWSKU                      ASIC       HEARTBEAT   APPLIED   APPLIEDG   CURRENTG   VERSION   SOFTWARE                ATTEMPT   ATTEMPTG   ROCE   CURRROCE   ECMPQPN   AGE
+as4630-01      server-leaf   leaf-4        Accton-AS4630-54NPE        broadcom   21s         31m       35         35         v0.84.3   4.5.0-Campus            31m       35                                     18d
+as7326-01      server-leaf   leaf-2        Accton-AS7326-56X          broadcom   15s         7m12s     46         46         v0.84.3   4.5.0-Enterprise_Base   7m12s     46         true   true                 18d
+as7712-01      spine         spine-1       Accton-AS7712-32X          broadcom   28s         37m       7          7          v0.84.3   4.5.0-Enterprise_Base   37m       7                                      18d
+as7712-02      spine         spine-2       Accton-AS7712-32X          broadcom   22s         17m       6          6          v0.84.3   4.5.0-Enterprise_Base   17m       6                                      18d
+s5248-05       server-leaf   leaf-1        DellEMC-S5248f-P-25G-DPB   broadcom   16s         85s       38         38         v0.84.3   4.5.0-Enterprise_Base   85s       38         true   true                 18d
+ds3000-01      server-leaf   leaf-5        DS3000                     broadcom   10s         23m       37         37         v0.84.3   4.5.0-Enterprise_Base   23m       37                                     18d
+```
+
+## Heart Beats
+
+### Diagnosing Long or Missing Heart beats
+
+If the duration between heart beats is greater than 1 minute, one of several
+things could be happening:
+
+* The connection between the switch and the control node is down or misconfigured.
+* The agent isn't running on the switches
+* The agent is stuck trying to apply a configuration
+
+#### Switch and Controller Connection
+
+The controller serves DHCP to the switches as part of normal operation and
+replies to ONIE requests for NOS install. The controller expects layer 2
+reachability and to be the only DHCP sever in the LAN. This architecture
+can take many forms but it is most often a dedicated switch or private VLAN.
+Ensure that the link on the controller is in the `up` state. Often problems
+in this area are related to the configuration of a private VLAN, or cabling.
+
+#### Starting and Stoping the Agent
+
+The switch agent is a daemon that is controlled via `systemctl` commands.
+The agent is started automatically at switch bootup time, to stop the agent:
+```bash
+sudo systemctl stop hedgehog-agent.service
+```
+similarly to start the agent:
+```bash
+sudo systemctl start hedgehog-agent.service
```

Comment:
Any reason why not just `sudo systemctl restart ...`?

Reply:
Occasionally, we will stop the agent so that debugging on the configuration can be done. The agent will re-write the configuration on a cyclic basis. So depending on how long the debugging takes we want the agent to not mess with the config while we are messing with the config.

Reply:
I see, why not explaining this in the docs? :slightly_smiling_face: